Src/Utils/S3Utils.cs

```diff
@@ -39,21 +40,37 @@ public static string GetImageHex(AmazonS3Client client, string bucket, string ke
             }
         }
 
-        public static Tile GetTile(AmazonS3Client client, string bucket, string path, int z, int x, int y)
+        public static Tile GetTileTMS(AmazonS3Client client, string bucket, string path, int z, int x, int y)
         {
-            Tile correspondingTile = null;
-            string key = $"{path}{z}/{z}/{y}.png";
+            Tile tile = null;
+            string key = $"{path}{z}/{x}/{y}.png";
+
             string blob = S3Utils.GetImageHex(client, bucket, key);
             if (blob != null)
             {
-                correspondingTile = new Tile(z, x, y, blob, blob.Length);
+                // We work with non-TMS tiles, so we convert y
+                y = GeoUtils.convertTMS(z, y);
+                tile = new Tile(z, x, y, blob, blob.Length);
             }
-            return correspondingTile;
+            return tile;
+        }
+
+        public static Tile GetTileTMS(AmazonS3Client client, string bucket, string path, Coord coord)
+        {
+            return GetTileTMS(client, bucket, path, coord.z, coord.x, coord.y);
+        }
+
+        public static Tile GetTile(AmazonS3Client client, string bucket, string path, int z, int x, int y)
```

Comment:
why not use Coord object too?

Reply:
There was no use for it in this context.
The places from which it was called already had a tile at hand or the values of z,x,y.